### PR TITLE
Show notification when tracking to create an itinerary

### PR DESCRIPTION
**Pull request by 276746**
# Changes:
When the user click on the "Start" button to track his position to create an itinerary, a notification is now displayed to show that he is currently being tracked. The notification disappear after clicking the "Stop" button.

# #
Screenshot 2024-06-02 180529

### CODE REVIEW

**Review by ethangraham2001 — APPROVED** [review]
Awesome stuff. Great work

File: app/src/main/java/com/github/wanderwise_inc/app/MainActivity.kt [hunk 1]
```diff
@@ -100,6 +102,20 @@ class MainActivity : ComponentActivity() {
...
+            finish()
+          }
+        }
+
+    if (checkSelfPermission(Manifest.permission.POST_NOTIFICATIONS) !=
+        PackageManager.PERMISSION_GRANTED) {
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+        launcher.launch(Manifest.permission.POST_NOTIFICATIONS)
+      }
+    }
```
Comment (YofiY) [inline app/src/main/java/com/github/wanderwise_inc/app/MainActivity.kt:118]: very clean great work !

File: app/src/main/java/com/github/wanderwise_inc/app/viewmodel/LocationService.kt [hunk 2]
```diff
@@ -1,12 +1,72 @@
...
+
+  /**
+   * A companion object is an object that is common to all instances of a class.
+   *
+   * This companion object contains constants that represent the actions that can be performed by
+   * the `LocationService`.
+   */
+  companion object {
+    const val ACTION_START = "ACTION_START"
+    const val ACTION_STOP = "ACTION_STOP"
```
Comment (YofiY) [inline app/src/main/java/com/github/wanderwise_inc/app/viewmodel/LocationService.kt:70]: necessary. Thank you Oscar!

**Review by YofiY — APPROVED** [review]
Overall very clean code, inspiring work!

**Pull request merged**